### PR TITLE
VectorDataset: extract mint/maxt from filename

### PR DESCRIPTION
**Pull request by oddeirikigland**
Opening PR after a discussion on Slack: https://torchgeo.slack.com/archives/C02DD9Q76SE/p1705589213060449

### CODE REVIEW

**Review by adamjstewart — CHANGES_REQUESTED** [review]
Can you copy https://github.com/microsoft/torchgeo/blob/v0.5.1/torchgeo/datasets/geo.py#L323-L339 from RasterDataset to VectorDataset? You can remove these lines: https://github.com/microsoft/torchgeo/blob/v0.5.1/torchgeo/datasets/geo.py#L329-L333

We'll also need to add tests for this. I can point you to where these would be, but let me know if you need help.

**Comment by oddeirikigland** [comment]
@microsoft-github-policy-service agree

**Comment by oddeirikigland** [comment]
> @oddeirikigland please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree

**Comment by oddeirikigland** [comment]
> Can you copy https://github.com/microsoft/torchgeo/blob/v0.5.1/torchgeo/datasets/geo.py#L323-L339 from RasterDataset to VectorDataset? You can remove these lines: https://github.com/microsoft/torchgeo/blob/v0.5.1/torchgeo/datasets/geo.py#L329-L333
> 
> We'll also need to add tests for this. I can point you to where these would be, but let me know if you need help.

Added the docstring. I'll need some guidance for the tests, where is the similar test for RasterDataset?

File: torchgeo/datasets/geo.py [hunk 1]
```diff
@@ -640,6 +654,10 @@ def __init__(
             else:
                 mint = 0
                 maxt = sys.maxsize
+                match = re.match(filename_regex, os.path.basename(filepath))
```
Comment (adamjstewart) [inline torchgeo/datasets/geo.py:657]: This needs a:
```python
if match:
```
right?

Comment (adamjstewart) [reply torchgeo/datasets/geo.py:657]: Let's move the re.match outside of the try-except, similar to RasterDataset. Then we can skip files that don't match the regex.

**Comment by adamjstewart** [comment]
For testing, you'll want to edit `tests/data/vector/data.py` such that the generated fake file includes a year in the filename. Then edit `CustomVectorDataset` in `tests/data/test_geo.py` to add a `filename_regex` and `date_format` to match.

File: torchgeo/datasets/geo.py (hunk 1 above)
Comment (oddeirikigland) [reply torchgeo/datasets/geo.py]: yep, fixed

**Comment by oddeirikigland** [comment]
> For testing, you'll want to edit `tests/data/vector/data.py` such that the generated fake file includes a year in the filename. Then edit `CustomVectorDataset` in `tests/data/test_geo.py` to add a `filename_regex` and `date_format` to match.

added test

**Review by adamjstewart — APPROVED** [review]
Looks pretty straightforward to me. Hopefully we'll get some builtin datasets that use this feature in the future. Not sure what kind of data your group is working with but would love to have it in TorchGeo if it's public. #1813 might actually use this feature too.

**Pull request merged**